src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/aarch64/CallArranger.java

```diff
@@ -197,14 +202,14 @@ static boolean isHomogeneousFloatAggregate(MemoryLayout type) {
         if (!(baseType instanceof ValueLayout))
             return false;
 
-        ArgumentClassImpl baseArgClass =
-            (ArgumentClassImpl)Utils.getAnnotation(baseType, ArgumentClassImpl.ABI_CLASS);
+        var optAbiType = baseType.abiType();
+        ArgumentClassImpl baseArgClass = optAbiType.isPresent()? AArch64ABI.argumentClassFor(optAbiType.get()) : null;
```

Comment:
Missing space before `?` Consider replacing with Optional::orElseThrow

Reply:
You can replace the ternary operator with `Optional::map`:
````java
optAbiType.map(AArch64ABI::argumentClassFor).orElse(null) // or orElseThrow() in case it should throw
````